src/featureManager.ts

```diff
@@ -30,15 +35,68 @@ export class FeatureManager {
 
     // If multiple feature flags are found, the first one takes precedence.
     async isEnabled(featureName: string, context?: unknown): Promise<boolean> {
-        const featureFlag = await this.#provider.getFeatureFlag(featureName);
-        if (featureFlag === undefined) {
-            // If the feature is not found, then it is disabled.
-            return false;
+        const result = await this.#evaluateFeature(featureName, context);
+        return result.enabled;
+    }
+
+    async getVariant(featureName: string, context?: ITargetingContext): Promise<Variant | undefined> {
+        const result = await this.#evaluateFeature(featureName, context);
+        return result.variant;
+    }
+
+    async #assignVariant(featureFlag: FeatureFlag, context: ITargetingContext): Promise<{
+        variant: VariantDefinition | undefined;
+        reason: VariantAssignmentReason;
+    }> {
+        // user allocation
+        if (featureFlag.allocation?.user !== undefined) {
+            for (const userAllocation of featureFlag.allocation.user) {
+                if (isTargetedUser(context.userId, userAllocation.users)) {
+                    const variant = featureFlag.variants?.find(v => v.name == userAllocation.variant);
+                    if (variant !== undefined) {
+                        return { variant, reason: VariantAssignmentReason.User };

```

Comment:
Should it be {variant: variant, reason: VariantAssignmentReason.User }?

Reply:
`{variant}` is short for `{variant: variant}`. A syntax sugar for simplicity, I can make it clearer if it looks confusing to you now.

Reply:
I am ok with it. But personally, I prefer consistency. Is there linting rule for this?

Reply:
Of course there is corresponding linting rule, it's called [object-shorthand](https://eslint.org/docs/latest/rules/object-shorthand).
This shorthand syntax makes the code more readable and is widely adopted in ES6 and later versions of JavaScript. See the example of this rule, usually we force to use object shorthand in linting rules instead of forbidding.